01-data-structures/02-stacks-and-queues/myqueue/myqueue.rb

```diff
@@ -8,11 +8,38 @@ def initialize
   end
 
   def enqueue(element)
+    if @queue.empty?
+      @queue[0] = element
+    else
+      new_array = [element]
+      @queue.each_with_index do |person, index|
+        new_array[index + 1] = @queue[index]
+      end
+      @queue = new_array
+    end
+    @head = @queue[0]
+    @tail = @queue[-1]
   end
 
   def dequeue
+    if @queue.empty?
+      return true
+    else
+      second_array = []
+      @queue.each_with_index do |person, index|
+        @queue[index - 1]
+      end
+      @queue = second_array
+    end
+    @tail = @queue[-1]
+    @head = @queue[0]
   end
 
   def empty?
+    if @queue.length == 0
+      return true
+    else
+      return false
+    end
```

Comment:
This is a common bit of logic that I correct. If you translate your statement to an English-based statement, it would sound like "If equating `queue`'s length to `0` yields the boolean `true`, then return the boolean `true`. If it yields `false`, then return the boolean `false`".

I will not reject the assignment due to this logic, but I would like to challenge you to determine what I would correct about it.